tests/integration/test_main.py

```diff
@@ -117,9 +117,9 @@ def test_clone__full_url_git(self):
         repo_slug_branch, seen_args = self.main_clone('git@service.com/guyzmo/git-repo', 0)
         assert ('guyzmo', 'git-repo', 'master') == repo_slug_branch
 
-    def test_clone__too_many_slashes(self):
-        did_clone = self.main_clone('guyzmo/git/repo', 2)
-        assert None is did_clone
+    # def test_clone__too_many_slashes(self):
+    #     did_clone = self.main_clone('guyzmo/git/repo', 2)
+    #     assert None is did_clone
```

Comment:
ok, I understand that this one is less obvious, and I'll make it a todo for me. So if you don't feel like it, leave it commented out, it's on me 😉

Basically, the way I'd implement it would be to add a new optional parameter to `self.main_clone()` that specifies the number of allowed backslashes, so that the test would be:

```python
def test_clone__not_too_many_slashes(self):
    did_clone = self.main_clone('guyzmo/git/repo', allowed_ns=3, 0)


def test_clone__too_many_slashes(self):
    did_clone = self.main_clone('guyzmo/git/repo', allowed_ns=1, 2)
```

where the `main_clone` helper would change the allowed value of `RepositoryService`'s mockup before calling the `do_clone()` function.